action.yaml

```diff
@@ -1,24 +1,151 @@
-name: action-setup-tools
+name: "Setup Multiple Language Environments"
 description:
-  GitHub Action that installs and provisions supported tools for workflow steps
-  in self-hosted runners. This relies on the agent having supported tooling
-  installed.
-outputs:
-  go:
-    description:
-      The version of golang that has been installed and is ready for use.
+  "Set up tools e.g Python, Node.js, Go, Java, and Terraform using standard
+  setup actions"
+
+inputs:
+  python:
+    description: "Python version (e.g. 3.11)"
+    required: false
   node:
-    description:
-      The verison of Node.js that has been installed and is ready for use.
+    description: "Node.js version (e.g. 18)"
+    required: false
+  go:
+    description: "Go version (e.g. 1.22)"
+    required: false
   java:
     description:
-      The version of Java that has been installed and is ready for use.
-  python:
-    description:
-      The version of Python that has been installed and is ready for use.
+      "Java version with distribution (e.g. 8.0.442-tem) which is the same as in
+      .sdkmanrc"
+    required: false
   terraform:
-    description:
-      The version of Terraform that has been installed and is ready for use.
+    description: "Terraform version (e.g. 1.6.6)"
+    required: false
+
 runs:
-  using: "node20"
-  main: "dist/main/index.js"
+  using: composite
+  steps:
+    - id: detect-python
+      shell: bash
+      run: |
+        if [ -n "${{ inputs.python }}" ]; then
+          echo "PYTHON_VERSION=${{ inputs.python }}" >> $GITHUB_ENV
+        elif [ -f .python-version ]; then
+          echo "PYTHON_VERSION=$(cat .python-version)" >> $GITHUB_ENV
+        else
+          echo "PYTHON_VERSION=" >> $GITHUB_ENV
+        fi
+
+    - name: Setup Python
+      if: env.PYTHON_VERSION != ''
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - id: detect-node
```

Comment:
We might not need to do this for node (and other languages as well), since the action supports specifying the node version and node version file. 

See https://github.com/actions/setup-node

<img width="809" alt="image" src="https://github.com/user-attachments/assets/eebd7fb0-4da2-4de5-8ecb-cc1b2d40cc8f" />

We might just need to pass `inputs.node` to `node-version` only and the action should just work